src/PubSub/Queue.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+namespace Myli\PlainJobs\PubSub;
+
+use Illuminate\Queue\Queue as IlluminateQueue;
+use Google\Cloud\PubSub\Topic;
+use Google\Cloud\PubSub\Message;
+use Google\Cloud\PubSub\PubSubClient;
+use Illuminate\Contracts\Queue\Queue as QueueContract;
+use Myli\PlainJobs\Jobs\DispatcherJob;
+use Illuminate\Support\Facades\Config;
+
+class Queue extends IlluminateQueue implements QueueContract
+{
+    /**
+     * The PubSubClient instance.
+     *
+     * @var \Google\Cloud\PubSub\PubSubClient
+     */
+    protected $pubsub;
+
+    /**
+     * Default queue name.
+     *
+     * @var string
+     */
+    protected $default;
+
+    /**
+     * Create a new GCP PubSub instance.
+     *
+     * @param \Google\Cloud\PubSub\PubSubClient $pubsub
+     * @param string                            $default
+     */
+    public function __construct(PubSubClient $pubsub, string $default = null)
+    {
+        $this->pubsub  = $pubsub;
+        $this->default = $default;
+    }
+
+    /**
+     * Get the size of the queue.
+     * PubSubClient have no method to retrieve the size of the queue.
+     * To be updated if the API allow to get that data.
+     *
+     * @param  string $queue
+     * @return int
+     */
+    public function size($queue = null)
+    {
+        return 0;
+    }
+
+    /**
+     * Push a new job onto the queue.
+     *
+     * @param  string|object $job
+     * @param  mixed         $data
+     * @param  string        $queue
+     * @return mixed
+     */
+    public function push($job, $data = '', $queue = null)
+    {
+        return $this->pushRaw($this->createPayload($job, $queue, $data), $queue);
+    }
+
+    /**
+     * Push a raw payload onto the queue.
+     *
+     * @param  string $payload
+     * @param  string $queue
+     * @param  array  $options
+     * @return array
+     */
+    public function pushRaw($payload, $queue = null, array $options = null)
+    {
+        $topic   = $this->getTopic($queue);
+        $publish = $topic->publish([
```

Comment:
You can `return ($topic->publish([
            'data'       => $payload,
            'attributes' => $options,
        ]));` as you only initialize `$publish` to return it. It means you allocate memory on the heap you will never use anyway and this also means garbage collection will have more work in the future. Only initialize a variable for a reason.